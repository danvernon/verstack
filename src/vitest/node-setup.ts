import { vi } from "vitest";

// Mock server-only package
vi.mock("server-only", () => {
  return {};
});

// Mock Clerk auth
vi.mock("@clerk/nextjs/server", () => ({
  auth: () => ({
    userId: "test-user-id",
    sessionId: "test-session-id",
    getToken: async () => "mock-token",
  }),
}));
