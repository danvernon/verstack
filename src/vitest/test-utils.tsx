import type { FC, ReactNode } from "react";
import { render as rtlRender } from "@testing-library/react";

const Providers: FC<{ children: ReactNode }> = ({ children }) => {
  return (
    <>
      {children}
      {/* Add any additional providers or context here */}
    </>
  );
};

const customRender = (ui: ReactNode, options = {}) => {
  return rtlRender(ui, { wrapper: Providers, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
