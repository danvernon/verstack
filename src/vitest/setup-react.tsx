// src/vitest/setup-react.ts
import NextImage from "next/image";
import { vi } from "vitest";

// Add React to the global scope for tests (needed for Next.js App Router components)
vi.mock("react", async () => {
  const actual = await vi.importActual("react");
  return {
    ...actual,
    // Add any specific React mock overrides here if needed
  };
});

// Mock common Next.js components and utilities
vi.mock("next/navigation", () => ({
  redirect: vi.fn(),
  useRouter: vi.fn(() => ({
    push: vi.fn(),
    replace: vi.fn(),
    refresh: vi.fn(),
    back: vi.fn(),
    prefetch: vi.fn(),
  })),
  usePathname: vi.fn(() => "/"),
  useSearchParams: vi.fn(() => new URLSearchParams()),
  useParams: vi.fn(() => ({})),
}));

// Mock Next.js image component
vi.mock("next/image", () => ({
  __esModule: true,
  default: ({
    src,
    alt,
    ...rest
  }: {
    src: string;
    alt: string;
    [key: string]: unknown;
  }) => {
    const Image = NextImage;
    return <Image src={src} alt={alt} {...rest} />;
  },
}));
