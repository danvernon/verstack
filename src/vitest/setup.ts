import "@testing-library/react";
import "@testing-library/jest-dom";

import React from "react";
import { cleanup } from "@testing-library/react";
import { beforeEach, vi } from "vitest";

beforeEach(() => {
  cleanup();
});

global.React = React;

if (typeof window === "undefined") {
  global.window = globalThis.window || Object.create(null);
}

global.window.matchMedia = function (query) {
  return {
    matches: false,
    media: query,
    onchange: null,
    addListener: vi.fn(),
    removeListener: vi.fn(),
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
    dispatchEvent: vi.fn(),
  };
};

vi.mock("next/navigation", () => ({
  useRouter: vi.fn().mockReturnValue({
    push: vi.fn(),
    replace: vi.fn(),
    back: vi.fn(),
    prefetch: vi.fn(),
  }),
  usePathname: vi.fn(() => "/mock-path"),
  useParams: vi.fn(() => ({ id: "mock-id" })),
}));
