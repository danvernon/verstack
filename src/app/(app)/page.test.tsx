import type { Mock } from "vitest";
import React from "react";
import { redirect } from "next/navigation";
import { auth } from "@clerk/nextjs/server";
import { afterEach, describe, expect, it, vi } from "vitest";

import Home from "./page";

// Mock React since Next.js App Router components use JSX without explicit imports
vi.mock("react", async () => {
  const actual = await vi.importActual("react");
  return {
    ...actual,
  };
});

vi.mock("@/trpc/server", () => ({
  api: {
    user: {
      getUser: vi.fn(),
      create: vi.fn(),
    },
    company: {
      get: vi.fn(),
    },
  },
}));

vi.mock("@clerk/nextjs/server", () => ({
  auth: vi.fn(),
}));

vi.mock("next/navigation", () => ({
  redirect: vi.fn(),
}));

// Mock any components imported by the page
vi.mock("@/components/dashboard", () => ({
  Dashboard: () => <div data-testid="dashboard">Dashboard Component</div>,
}));

vi.mock("@/components/company/create-company", () => ({
  default: () => (
    <div data-testid="create-company">Create Company Component</div>
  ),
}));

describe("Home", () => {
  afterEach(() => {
    vi.clearAllMocks();
  });

  it("redirects to /sign-in if userId is not present", async () => {
    (auth as unknown as Mock).mockResolvedValue({ userId: null });

    await Home();

    expect(redirect).toHaveBeenCalledWith("/sign-in");
  });
});
