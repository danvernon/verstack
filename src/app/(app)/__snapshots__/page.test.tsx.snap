// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > renders CreateCompany if no company exists 1`] = `
<div
  className="flex w-full justify-center"
>
  <default />
</div>
`;

exports[`Home > renders Dashboard if a company exists 1`] = `
<div
  className="flex w-full justify-center"
>
  <div
    className="rounded-lg bg-red-50 p-6 text-red-800"
  >
    <h2
      className="text-lg font-semibold"
    >
      Something went wrong
    </h2>
    <p>
      We're having trouble loading your data. Please try again later.
    </p>
  </div>
</div>
`;

exports[`Home > renders error message if an exception occurs 1`] = `
<div
  className="flex w-full justify-center"
>
  <div
    className="rounded-lg bg-red-50 p-6 text-red-800"
  >
    <h2
      className="text-lg font-semibold"
    >
      Something went wrong
    </h2>
    <p>
      We're having trouble loading your data. Please try again later.
    </p>
  </div>
</div>
`;
